### PR TITLE
Jetpack CLI: allow modification of all options, require confirmation for some

### PR DESCRIPTION
**Pull request by dereksmart**
Before, the CLI straight up prohibited modification of certain options.  This will allow the modification of any Jetpack option, but throws a confirmation prompt as a warning when modifying dangerous ones.  

`Are you sure? Modifying this option may disrupt your Jetpack connection.  Type "yes" to continue.`

To test, try to delete or update one of the "unsafe" options, like `version`.

### CODE REVIEW

File: class.jetpack-cli.php [hunk 1]
```diff
@@ -576,7 +569,13 @@ function jetpack_cli_are_you_sure( $error_msg = false ) {
 		$error_msg = sprintf( __( 'Action cancelled. Have a question? %sjetpack.me/support%s', 'jetpack' ), $cli->green_open, $cli->color_close );
 	}
 
-	WP_CLI::line( _x( 'Are you sure? This cannot be undone. Type "yes" to continue:', '"yes" is a command.  Do not translate that.', 'jetpack' ) );
+	if ( ! $flagged ) {
+		$prompt_message = __( 'Are you sure? This cannot be undone. Type "yes" to continue:', '"yes" is a command.  Do not translate that.', 'jetpack' );
+	} else {
+		$prompt_message = __( 'Are you sure? Deleting this option may disconnect your site from jetpack.  Type "yes" to continue.', 'jetpack' );
```
Comment (kraftbj) [inline class.jetpack-cli.php]: s/Deleting/Modifying? Or "Modifying or Deleting" etc

File: class.jetpack-cli.php [hunk 2]
```diff
@@ -459,6 +459,7 @@ public function protect( $args, $assoc_args ) {
 	public function options( $args, $assoc_args ) {
 		$action = isset( $args[0] ) ? $args[0] : 'list';
 		$safe_to_modify = Jetpack::get_jetapck_options_for_reset();
+		$flagged = !in_array( $args[1], $safe_to_modify ) ? true : false;
```
Comment (zinigor) [inline class.jetpack-cli.php]: I think you meant to put this after you push `jumpstart` to `$safe_to_modify`.

File: class.jetpack-cli.php [hunk 3]
```diff
@@ -459,6 +459,7 @@ public function protect( $args, $assoc_args ) {
 	public function options( $args, $assoc_args ) {
 		$action = isset( $args[0] ) ? $args[0] : 'list';
 		$safe_to_modify = Jetpack::get_jetapck_options_for_reset();
```
Comment (zinigor) [inline class.jetpack-cli.php]: Can you please fix `jetapck` everywhere?

File: class.jetpack.php [hunk 4]
```diff
@@ -5824,11 +5825,17 @@ public static function get_jetapck_options_for_reset() {
...
 			'activated',
+
+			// private
+			'register',
+			'blog_token',                  // (string) The Client Secret/Blog Token of this site.
+			'user_token',                  // (string) The User Token of this site. (deprecated)
+			'user_tokens'
 		);
 
 		// Remove the whitelisted Jetpack options
```
Comment (zinigor) [inline class.jetpack.php]: Nitpick, but now that you have renamed them to `unsafe`, this comment is confusing.

**Comment by zinigor** [comment]
Very nice overall, needs some minor fixes.

**Comment by dereksmart** [comment]
made the requested edits.  thanks!

**Comment by zinigor** [comment]
Tested, works as expected!

**Pull request merged**